pandas/tests/arithmetic/test_object.py

```diff
@@ -281,13 +282,14 @@ def test_add(self):
 
     def test_sub_fail(self):
         index = tm.makeStringIndex(100)
-        with pytest.raises(TypeError):
+        msg = "unsupported operand type|<class 'list'>"  # Error for index.toList()-list is uninformative?
+        with pytest.raises(TypeError, match=msg):
             index - "a"
-        with pytest.raises(TypeError):
+        with pytest.raises(TypeError, match=msg):
             index - index
-        with pytest.raises(TypeError):
+        with pytest.raises(TypeError, match=msg):
             index - index.tolist()
-        with pytest.raises(TypeError):
+        with pytest.raises(TypeError, match=msg):
             index.tolist() - index
```

Comment:
The error message given by this code is  just `<class 'list'>`.  Am I correct that this error message is not sufficiently informative?  Is this an issue?

Reply:
Yes, this is not informative. Instead of testing a wrong error message, I would rather fix or add a TODO comment here about needing to improve it.